### ISSUE TITLE
Error while cleaning gitlab webhooks

### ISSUE THREAD

**Issue by FlorianBorn** [MicroFocus/octane-gitlab-service#102]
Hi all,

we want to cleanup our gitlab webhooks on a regular basis and receive an error while running the octane-gitlab-service with cleanupOnly flag. The webhook is removed but the service is stuck.

We would expect the service to remove the webhooks and to exit afterwards
 
The command we use
`java -jar octane-gitlab-service.jar --cleanupOnly=true`

Version: 23.3.2

<details>
  <summary>The error</summary>  

```
[INFO ] 2023-11-16 08:20:59.981 [background-preinit] Version - HV000001: Hibernate Validator 6.2.5.Final
[INFO ] 2023-11-16 08:21:00.049 [main] Application - Starting Application v23.3.2 using Java 17.0.9 on c34333c36b90 with PID 13 (/octane_gitlab_service/octane-gitlab-service-23.3.2.jar started by root in /octane_gitlab_service)
[INFO ] 2023-11-16 08:21:00.053 [main] Application - No active profile set, falling back to 1 default profile: "default"
[INFO ] 2023-11-16 08:21:00.913 [main] log - Logging initialized @2253ms to org.eclipse.jetty.util.log.Slf4jLog
[INFO ] 2023-11-16 08:21:01.055 [main] JettyServletWebServerFactory - Server initialized with port: 9091
[INFO ] 2023-11-16 08:21:01.058 [main] Server - jetty-9.4.51.v20230217; built: 2023-02-17T08:19:37.309Z; git: b45c405e4544384de066f814ed42ae3dceacdd49; jvm 17.0.9+8-LTS
[INFO ] 2023-11-16 08:21:01.107 [main] application - Initializing Spring embedded WebApplicationContext
[INFO ] 2023-11-16 08:21:01.107 [main] ServletWebServerApplicationContext - Root WebApplicationContext: initialization completed in 997 ms
[INFO ] 2023-11-16 08:21:01.300 [main] session - DefaultSessionIdManager workerName=node0
[INFO ] 2023-11-16 08:21:01.300 [main] session - No SessionScavenger set, using defaults
[INFO ] 2023-11-16 08:21:01.301 [main] session - node0 Scavenging every 600000ms
[INFO ] 2023-11-16 08:21:01.309 [main] ContextHandler - Started o.s.b.w.e.j.JettyEmbeddedWebAppContext@4b2a01d4{application,/,[file:///tmp/jetty-docbase.9091.11360211993521146769/],AVAILABLE}
[INFO ] 2023-11-16 08:21:01.310 [main] Server - Started @2652ms
[INFO ] 2023-11-16 08:21:03.095 [main] GitlabServices - start with cleanup process
[INFO ] 2023-11-16 08:21:04.252 [main] AbstractConnector - Started ServerConnector@132e0cc{HTTP/1.1, (http/1.1)}{0.0.0.0:9091}
[INFO ] 2023-11-16 08:21:04.253 [main] JettyWebServer - Jetty started on port(s) 9091 (http/1.1) with context path '/'
[INFO ] 2023-11-16 08:21:04.262 [main] Application - Started Application in 4.548 seconds (JVM running for 5.605)
[INFO ] 2023-11-16 08:21:04.666 [main] GitlabServices - Success while accessing the 'https://foobar.de/events' endpoint.
clean-up webhooks from Gitlab server process is finished. Stopping the service
[INFO ] 2023-11-16 08:21:04.703 [main] AbstractConnector - Stopped ServerConnector@132e0cc{HTTP/1.1, (http/1.1)}{0.0.0.0:9091}
[INFO ] 2023-11-16 08:21:04.704 [main] session - node0 Stopped scavenging
[INFO ] 2023-11-16 08:21:04.710 [main] ContextHandler - Stopped o.s.b.w.e.j.JettyEmbeddedWebAppContext@4b2a01d4{application,/,[file:///tmp/jetty-docbase.9091.11360211993521146769/],STOPPED}
[INFO ] 2023-11-16 08:21:04.722 [main] GitlabServices - stop Executors
[WARN ] 2023-11-16 08:21:04.722 [main] GitlabServices - Failed to destroy GitLab webhooks
java.lang.NullPointerException: Cannot invoke "java.util.concurrent.ScheduledFuture.cancel(boolean)" because "this.updateHooksScheduledFuture" is null
        at com.microfocus.octane.gitlab.services.GitlabServices.stopExecutors(GitlabServices.java:169) ~[classes!/:23.3.2]
        at com.microfocus.octane.gitlab.services.GitlabServices.stop(GitlabServices.java:151) ~[classes!/:23.3.2]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
        at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:389) ~[spring-beans-5.3.27.jar!/:5.3.27]
        at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeDestroyMethods(InitDestroyAnnotationBeanPostProcessor.java:347) ~[spring-beans-5.3.27.jar!/:5.3.27]
        at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeDestruction(InitDestroyAnnotationBeanPostProcessor.java:177) ~[spring-beans-5.3.27.jar!/:5.3.27]
        at org.springframework.beans.factory.support.DisposableBeanAdapter.destroy(DisposableBeanAdapter.java:197) ~[spring-beans-5.3.27.jar!/:5.3.27]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroyBean(DefaultSingletonBeanRegistry.java:587) ~[spring-beans-5.3.27.jar!/:5.3.27]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingleton(DefaultSingletonBeanRegistry.java:559) ~[spring-beans-5.3.27.jar!/:5.3.27]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingleton(DefaultListableBeanFactory.java:1163) ~[spring-beans-5.3.27.jar!/:5.3.27]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingletons(DefaultSingletonBeanRegistry.java:520) ~[spring-beans-5.3.27.jar!/:5.3.27]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingletons(DefaultListableBeanFactory.java:1156) ~[spring-beans-5.3.27.jar!/:5.3.27]
        at org.springframework.context.support.AbstractApplicationContext.destroyBeans(AbstractApplicationContext.java:1108) ~[spring-context-5.3.27.jar!/:5.3.27]
        at org.springframework.context.support.AbstractApplicationContext.doClose(AbstractApplicationContext.java:1077) ~[spring-context-5.3.27.jar!/:5.3.27]
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.doClose(ServletWebServerApplicationContext.java:174) ~[spring-boot-2.7.12.jar!/:2.7.12]
        at org.springframework.context.support.AbstractApplicationContext.close(AbstractApplicationContext.java:1023) ~[spring-context-5.3.27.jar!/:5.3.27]
        at com.microfocus.octane.gitlab.app.Application.main(Application.java:84) ~[classes!/:23.3.2]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[octane-gitlab-service-23.3.2.jar:23.3.2]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:108) ~[octane-gitlab-service-23.3.2.jar:23.3.2]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[octane-gitlab-service-23.3.2.jar:23.3.2]
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65) ~[octane-gitlab-service-23.3.2.jar:23.3.2]
</details>

**Comment by TiutiuMadalin** [comment]
Hi again @FlorianBorn, I took a look some threads were still running, that's why the service remained stuck.
The fix I found is to not start those when in cleanup mode.

**Comment by TiutiuMadalin** [comment]
fixed with [v23.4.0-CE](https://github.com/MicroFocus/octane-gitlab-service/releases/tag/v23.4.0-CE)
